### PR TITLE
Feature/cmake

### PR DESCRIPTION
**Pull request by jcelerier**
Theo,

Would it be possible to merge feature/cmake into dev (at least for Score for now ? ) 

This is because I have to set up continuous integration for i-score and we agreed with Pascal that I should use the master branch of Score, however it won't build on Linux/Windows. So a prerequisite is to merge cmake in dev.

However please test it a bit before (it should work with JamomaCore on dev branch w/ the ruby scripts), because I had to change a bunch of stuff in PetriNet and TTCurve for it to work in windows.
Basically : Petrinet would not compile because it looks like there is another `Arc` class in the windows headers that messed things, so I renamed it to `PetriNetArc`. For TTCurve I had to transform the inheritance relationship with TTList into an encapsulation mechanism, because MSVC was not able to infer that TTCurve was a TTObject anymore, and it would refuse to compile because of the attribute registering.

Thanks!

### CODE REVIEW

**Comment by theod** [comment]
is it possible to do this wednesday of the coming week when we are all in the same place ? or is it urgent ?

**Comment by jcelerier** [comment]
Next week will be fine! Thanks!
